src/main.rs

```diff
@@ -131,62 +123,126 @@ enum SenderEvent {
     Listening(u16),
 }
 
+fn parse_socket_addr(s: String) -> std::result::Result<SocketAddr, String> {
+    SocketAddr::from_str(&s).map_err(|e| format!("Invalid address '{}': {}", s, e))
+}
+
+fn cli() -> OptionParser<Command> {
+    let verbosity = bpaf::short('v')
+        .long("verbose")
+        .help("Enable verbose debug output")
+        .flag(Verbosity::Verbose, Verbosity::Silent);
+
+    let send_parser = {
+        let max_bandwidth = long("max-bandwidth-mbps")
+            .help("Specify the maximum bandwidth to use over a 1 second sliding window, in MB/s. If unspecified, there will be no limit")
+            .argument::<u64>("MBPS")
+            .optional();
+
+        let listen_addr = positional::<String>("LISTEN_ADDR")
+            .help("Address (IP and port) for the sending side to bind to and listen for receivers. This should be the address of a Wireguard interface if you care about confidentiality. E.g. '100.71.154.83:7999'")
+            .parse(parse_socket_addr);
+
+        let fnames = positional::<String>("FILES")
+            .help("Paths of files to send. Input file paths need to be relative. This is a safety measure to make it harder to accidentally overwrite files in /etc and the like on the receiving end")
+            .some("At least one file must be specified");
+
+        construct!(Command::Send {
+            verbosity(verbosity.clone()),
```

Comment:
I would have verbosity as a function and call it here instead of using this more obscure syntax:

```rust
    fn verbosity() -> impl Parser<Verbosity> {
        bpaf::short('v')
        .long("verbose")
        .help("Enable verbose debug output")
        .flag(Verbosity::Verbose, Verbosity::Silent)
    }
```

and use it s 

```rust
construct!(Command::Send {
    verbosity(),
    max_bandwidth,
    ...
}
```

same below for Recv